fml/platform/darwin/scoped_nsautorelease_pool.cc

```diff
@@ -0,0 +1,22 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/platform/darwin/scoped_nsautorelease_pool.h"
+
+extern "C" {
+void* objc_autoreleasePoolPush(void);
```

Comment:
These are undocumented methods.  Can you just turn this to a .mm file and use the objc objects?

Reply:
Alternatively, just rename the playground test file to `golden_playground_test_mac.mm` and create and destroy the pool in the SetUp and TearDown (storing the pool in `GoldenPlaygroundTest::GoldenPlaygroundTestImpl`).

Reply:
This needs to be importable into non-objc TUs See https://github.com/flutter/engine/pull/40748/files for example

Reply:
> This needs to be importable into non-objc TUs See https://github.com/flutter/engine/pull/40748/files for example

GoldenPlaygroundTest already handles this by using the pimpl pattern.  For `PlaygroundTest` it can just remain a `void*` which is no worse than what we have now.

Reply:
This is based on Chromium's usage of these APIs in a similar class: see https://source.chromium.org/chromium/chromium/src/+/main:base/mac/scoped_nsautorelease_pool.cc

Relying on implementation details of the ObjC runtime is not ideal, but I don't think it will cause issues as long as it's hidden behind the `ScopedNSAutoreleasePool` abstraction.

Reply:
Okay, we keep the undocumented functions can you add this file to a `testonly` build target?  On iOS using undocumented functions can result in getting rejected from the App Store.  We want to make sure that it doesn't find it's way into the iOS code.